File: DownloadsPdfsRobo.js
const fs = require('fs');
const path = require('path');
const https = require('https');
const http = require('http');

// Function to create directory if it doesn't exist
const createDirectoryIfNotExists = (directory) => {
    if (!fs.existsSync(directory)) {
        fs.mkdirSync(directory, { recursive: true });
    }
};

// Function to download a file from a URL
const downloadFile = (url, filePath) => {
    const protocol = url.startsWith('https') ? https : http;
    return new Promise((resolve, reject) => {
        protocol.get(url, (response) => {
            const fileStream = fs.createWriteStream(filePath);
            response.pipe(fileStream);
            fileStream.on('finish', () => {
                fileStream.close();
                resolve();
            });
            fileStream.on('error', (err) => {
                fs.unlink(filePath, () => reject(err));
            });
        }).on('error', (err) => {
            fs.unlink(filePath, () => reject(err));
        });
    });
};

// Main function to read JSON file and initiate downloads
const main = async () => {
    try {
        const folderPath = './arquivos/2015';
        createDirectoryIfNotExists(folderPath);

        const jsonFilePath = './arquivos/dissertação-2015.json';
        const jsonData = fs.readFileSync(jsonFilePath, 'utf8');
        const urlList = JSON.parse(jsonData);

        for (const url of urlList) {
            const fileName = path.basename(url);
            const filePath = path.join(folderPath, fileName);
            await downloadFile(url, filePath);
            console.log(`File '${fileName}' downloaded successfully.`);
        }

        console.log('All files downloaded successfully.');
    } catch (error) {
        console.error(`An error occurred: ${error.message}`);
    }
};

// Run the main function
main();

File: DownloadsPdfsRobo.py
import os
import json
import requests
from urllib.parse import urlparse

# Create directory to save files if it doesn't exist
folder_path = './arquivos/2015'
os.makedirs(folder_path, exist_ok=True)

# Path to the JSON file
json_file_path = './arquivos/dissertação-2015.json'

# Function to download files
def download_file(url, file_name):
    try:
        response = requests.get(url)
        with open(file_name, 'wb') as file:
            file.write(response.content)
        print(f"File '{file_name}' downloaded successfully.")
    except Exception as e:
        print(f"Error downloading file '{file_name}': {str(e)}")

# Read the JSON file
try:
    with open(json_file_path, 'r') as file:
        url_list = json.load(file)
except Exception as e:
    print(f"Error reading JSON file: {str(e)}")
    url_list = []

# Download files from URLs
for url in url_list:
    file_name = os.path.join(folder_path, os.path.basename(urlparse(url).path))
    download_file(url, file_name)
